app/assets/javascripts/lib/util/analytics.js

```diff
@@ -0,0 +1,31 @@
+var Analytics = (function () {
+    'use strict';
+
+    function track(obj) {
+        if (window.ga) {
+            window.ga('send', obj.event, obj.type, obj.data);
+        }
+    }
+
+    Analytics.prototype = {
+        chartAdded: function (key) {
+            track({
+                event: 'event',
+                type: 'chart-add',
+                data: key
+            });
+        },
+
+        changeInput: function (e) {
```

Comment:
Thinking about this a bit, should we really track *every* time a user touches an input? On the one hand it might show which sliders users spend the most time on (assuming that more changes = more time), but it might also drown out other interesting information like "how often is slider B used versus slider A"?

If we want to track that a user has touched a slider *one or more times* rather than logging an event for *each* change, [underscore's memoize](http://underscorejs.org/#memoize) can do the trick:

```js
function Analytics () {
  this.changeInput = _.memoize(this.changeInput);
}
```

If we do that, maybe the event needs to be called something else. "used input", "modified input", "touched input", ...?

@grdw @ChaelKruip Curious as to your thoughts.

---

Ditto for the charts. Memoizing would avoid the duplicate event being sent during the initial page load and would mean we only send an event the first time each chart is loaded.

Reply:
> Thinking about this a bit, should we really track every time a user touches an input?

So in other words - it's:

Checking which sliders are touched at all (regardless if the user drags it 1,2,3,4...400 times) **or** checking every time the slider is touched. I'm not sure what we want to track, so I don't really feel to strongly about this one.

Reply:
I actually discussed this last monday in short with @ChaelKruip and @jorisberkhout and it seems that only tracking which one's are used/touched is fine. As opposed to tracking everytime it has actually changed.